platform/docs/docs/platform/services/data/HangingProtocolService.md

```diff
@@ -27,38 +27,124 @@ registered automatically to the HangingProtocolService.
 
 All protocols are stored in the `HangingProtocolService` using their `id` as the key, and the protocol itself as the value.
 
+## Protocol Definition
+Protocols are defined in a getHangingProtocolModule inside an extension.  As such,
+they are defined with a module structure that starts with an id, and has field protocol
+that is the actual protocol definition.  This setup allows defining more than
+one protocol within a module, each one needing it's own definition file.
+
+```javascript
+import MyProtocol from './MyProtocol';
+export default function getHangingProtocolModule() {
+  return [
+    {
+      id: MyProtocol.id,
+      protocol: MyProtocol,
+    },
+  ];
+}
+```
+
+Within the protocol itself, the structure is layed out as described in the HangingProtocol.ts
+type definition, starting with `Protocol`.  See the type definition for more details.
+
 ## Events
 
 There are two events that get publish in `HangingProtocolService`:
 
 | Event        | Description                                                          |
 | ------------ | -------------------------------------------------------------------- |
 | NEW_LAYOUT   | Fires when a new layout is requested by the `HangingProtocolService` |
-| STAGE_CHANGE | Fires when the the stage is changed in the hanging protocols         |
-| PROTOCOL_CHANGED | Fires when the the protocol is changed in the hanging protocols         |
-| HANGING_PROTOCOL_APPLIED_FOR_VIEWPORT | Fires when the hanging protocol applies for a viewport (sets its displaySets) |
+| PROTOCOL_CHANGED | Fires when the the protocol is changed in the hanging protocols, or when the applied stage is changed. |
+| RESTORE_PROTOCOL | Fires when the protocol or stage is restored, for example, after turning off MPR mode |
+| STAGE_ACTIVATION | Fires when the stages are known to have stage.status set. |
+
+## Stage Activation and Status
+Sometimes a hanging protocol can be applicable generally, but not all stages
+should be shown by default, or should be shown at all.  This can be handled by
+using the stage activation to control whether the stage is shown by default (`enabled`),
+whether it can be navigated to (`passive`) or whether it should not be shown
+at all (`disabled`).
+
+The `stage.status` is used to control this, and the status is controlled by
+the stage activate.  The status values are:
+
+* enabled - meaning that the stage is fully applicable
+* passive - meaning that the stage can be applied, but might be missing details
+* disabled - meaning that the study has insuffient information for this stage
+
+The setting for these are controlled by the stageActivation property, for example
+
+```javascript
+stageActivation: {
```

Comment:
It would be good to say if you don't specify `stageActivation` at all, what is the default behaviour